great_expectations/data_context/types/base.py

```diff
@@ -434,6 +951,442 @@ def validate_schema(self, data, **kwargs):
+
+    compute_and_persist_to_s3_data_context_id(
+        project_config=project_config,
+        bucket=ge_id_config_bucket,
+        prefix=ge_id_config_prefix,
+        runtime_environment=None,
+        **ge_id_config_kwargs,
+    )
+
+    store_config: dict = {
+        "bucket": expectations_store_bucket,
+        "prefix": expectations_store_prefix,
+    }
+    store_config.update(**expectations_store_kwargs)
+    s3_store_backend_obj = build_tuple_s3_store_backend(**store_config)
+    project_config.add_expectations_store(
+        name=expectations_store_name, store_backend=s3_store_backend_obj
+    )
+    project_config.expectations_store_name = expectations_store_name
+
+    project_config_store.save_configuration(configuration=project_config)
+
+    return {
+        "project_config_store": project_config_store,
+        "new_project_config_created": new_project_config_created,
+    }
+
+
+# noinspection pyargumentlist
+@create_using_s3_backend
+def create_standard_s3_backend_project_config(**kwargs,):
+    func_callee: Callable = build_s3_backend_project_config
+    # noinspection SpellCheckingInspection
+    argspec: list = getfullargspec(func_callee)[0]
+    filter_properties_dict(
+        properties=kwargs, keep_fields=argspec, clean_empty=False, inplace=True
+    )
+    return func_callee(**kwargs)
+
+
+# noinspection SpellCheckingInspection
+def build_s3_backend_project_config(
+    datasource_type: str,
+    validations_store_bucket: str = None,
+    validations_store_prefix: str = None,
+    data_docs_store_bucket: str = None,
+    data_docs_store_prefix: str = None,
+    validations_store_name: str = None,
+    data_docs_site_name: str = "data_docs_site",
+    validations_store_kwargs: dict = None,
+    data_docs_store_kwargs: dict = None,
+    slack_webhook: str = None,
+    show_how_to_buttons: bool = True,
+    show_cta_footer: bool = True,
+    include_profiling: bool = True,
+    project_config_store=None,
+):
+    if project_config_store is None:
+        raise ge_exceptions.DataContextError(
+            f"""The build_s3_backend_project_config method requires a valid project_config_store reference.
+            """
+        )
+    if validations_store_bucket is None:
+        validations_store_bucket = ""
+    if validations_store_prefix is None:
+        validations_store_prefix = ""
+    if validations_store_kwargs is None:
+        validations_store_kwargs = {}
+    if data_docs_store_bucket is None:
+        data_docs_store_bucket = ""
+    if data_docs_store_prefix is None:
+        data_docs_store_prefix = ""
+    if data_docs_store_kwargs is None:
+        data_docs_store_kwargs = {}
+
+    project_config = project_config_store.load_configuration()
+
+    # Add the Data Source (Spark Dataframe and Pandas Dataframe data sources types are currently implemented).
+    if datasource_type == "spark":
+        spark_df_datasource_name: str = "s3_files_spark_datasource"
```

Comment:
We talked about it -- even though this would be my individual preference and default usecase, I am hesitant to force the users to Spark -- simply because while on Support duty, I have seen quite a few Pandas in S3 experiments that users reported.

Reply:
That totally makes sense. I was referring more to the function signature for `build_s3_backend_project_config` having two additional parameters: `spark_df_datasource_name` and `pandas_datasource_name` with the defaults you have here just so they are user configurable. So the user can do something like `build_s3_backend_project_config(datasource_type="spark", spark_df_datasource_name="my_spark_datasource_name")` or just `build_s3_backend_project_config(datasource_type="spark")`

Reply:
Actually even better - it could just be a single optional parameter `datasource_name` and if set then use it -- if not then use these defaults.